migration-from-4x.md

```diff
@@ -0,0 +1,328 @@
+* The `Ctrl+K` shortcut does nothing until the context toolbar is visible in TinyMCE 4.x.
+* In TinyMCE 5.0, the pop animates to its new width.
+* In TinyMCE 4.x., it is a URL input, so you get a popup and a browse button.
+
+### Custom Context Menus
+
+The default context menu is no longer a plugin, it is part of the core and always enabled. Where TinyMCE 4.x only supported adding registered menu items, the new context menu also allows plugins to register "sections" of the context menu. These sections are dynamic and may show or hide depending on the cursor position when the context menu is opened. For example, the default context menu config is now `'link image editimage table spelling'` which are all plugin section references, not menu items.
+
+##### New methods:
+
+| **New Method** | **Description** |
+| ----------- | -------------- |
+| editor.ui.registry.addContextMenu() | Adds a custom context menu. |
+
+For more information on Context Menus, see the [docs]({{site.baseurl}}/components/contextmenu).
+
+### Custom Context Forms
+
+ContextForms are a generalisation of the `Insert Link` form that existed in the original `inlite` plugin from [TinyMCE 4.x]((https://www.tiny.cloud/docs/themes/inlite/#quicklink)).
+
+### Toolbar Menus
+
+* New buttons are added to the global `editor.settings.menus` which is an enhancement since, it now shows the toggled state.
+* Improved mouse and keyboard navigation.
+
+## Plugins
+
+Each release of TinyMCE adds new features and functionality. We also occasionally remove features and functionality, usually because we've added a better option.
+Here are the details about the features and functionalities that we removed in TinyMCE 5.0.
+
+### Removed Features
+
+Plugins with dialogs no longer have `height` or `width` settings for their dialogs. The dialog component now uses CSS3 and a predefined `small`, `medium`, and `large` template to specify the dimensions.
+
+The following plugins from TinyMCE 4.x do not require height or width options to be specified in TinyMCE 5.0:
+
+* [Code]({{site.baseurl}}/plugins/code/)
+* [Codesample]({{site.baseurl}}plugins/codesample/)
+* [Preview]({{site.baseurl}}plugins/preview/)
+* [Template]({{site.baseurl}}plugins/template/)
+
+Please read the [docs]({{site.baseurl}}/plugins/) if you need more information on the Tiny 4.x configuration options.
+
+### Changed Features
+
+These features have either changed or have been deleted in TinyMCE 5.0.
+Each release of TinyMCE adds new features and functionality. We also occasionally remove features and functionality, usually because we’ve added a better option. Here are the details about the features and functionalities that we removed in TinyMCE 5.0.
+
+#### Moved Plugins
+
+| **Plugin Name** | **Description** |
+| --------------- |  -------------- |
+| [ContextMenu](https://www.tiny.cloud/docs/plugins/contextmenu/) | New API. See [docs]({{site.baseurl}}/components/contextmenu/). |
```

Comment:
I think we need to add colorpicker to this somewhere? @EkimChau ?

Reply:
Also textcolor - see DOC-143

Reply:
Added colorpicker to the list. Will add textcolor after Beta.